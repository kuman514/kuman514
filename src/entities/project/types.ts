export interface ProjectItem {
  projectId: string;
  title: string;
  thumbnailUrl: string;
  description: string;
  techStackTags: string[];
}
