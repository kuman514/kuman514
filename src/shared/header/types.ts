export interface NavNode {
  id: string;
  label: string;
  href: string;
}
