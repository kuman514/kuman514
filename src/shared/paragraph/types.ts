import { ReactNode } from 'react';

export interface ParagraphProps {
  customClassName?: string;
  children?: ReactNode;
}
