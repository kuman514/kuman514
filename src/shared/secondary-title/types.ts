import { ReactNode } from 'react';

export interface SecondaryTitleProps {
  customClassName?: string;
  children?: ReactNode;
}
