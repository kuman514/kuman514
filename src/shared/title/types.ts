import { ReactNode } from 'react';

export interface TitleProps {
  customClassName?: string;
  children?: ReactNode;
}
