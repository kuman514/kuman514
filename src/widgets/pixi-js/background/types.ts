import { Sprite } from 'pixi.js';

export interface IconComet {
  sprite: Sprite;
  scale: number;
  speed: number;
}
