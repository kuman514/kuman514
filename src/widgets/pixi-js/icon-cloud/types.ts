import { Sprite } from 'pixi.js';

export interface IconCloudIcon {
  sprite: Sprite;
  x: number;
  y: number;
  z: number;
}
