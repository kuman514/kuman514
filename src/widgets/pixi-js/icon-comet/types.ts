import { Sprite } from 'pixi.js';

export interface IconComet {
  sprite: Sprite;
  scale: number;
  speed: number;
}

export interface CanvasSize {
  canvasWidth: number;
  canvasHeight: number;
}

export interface RandomizeIconCometPositionParams extends CanvasSize {
  icon: IconComet;
  isInitial: boolean;
}
