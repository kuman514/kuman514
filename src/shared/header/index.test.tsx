import { fireEvent, render, screen } from '@testing-library/react';

import Header from '.';
import { NavNode } from './types';

const testNavNodes: NavNode[] = [
  {
    id: 'yasuo',
    label: 'Yasuo',
    href: '/yasuo',
  },
  {
    id: 'koishi',
    label: 'Koishi Komeiji',
    href: '/koishi',
  },
  {
    id: 'hoshino',
    label: 'Hoshino Takanashi',
    href: '/hoshino',
  },
];

const mockFn = jest.fn();

function MockedLink({
  children,
  href,
  'aria-label': ariaLabel,
}: React.AnchorHTMLAttributes<HTMLAnchorElement>) {
  return (
    <a
      aria-label={ariaLabel}
      onClick={() => {
        mockFn(href);
      }}
    >
      {children}
    </a>
  );
}

jest.mock('next/link', () => MockedLink);

describe('Header', () => {
  it('should have appearances that matches snapshot', async () => {
    const { container } = render(<Header navNodes={testNavNodes} />);
    expect(container.firstChild).toMatchSnapshot();
  });

  it('should have nodes that have working links', async () => {
    render(<Header navNodes={testNavNodes} />);

    fireEvent.click(await screen.findByLabelText('kuman514-logo'));
    expect(mockFn).toHaveBeenLastCalledWith('/');

    fireEvent.click(await screen.findByText(/Yasuo/i));
    expect(mockFn).toHaveBeenLastCalledWith('/yasuo');

    fireEvent.click(await screen.findByText(/Koishi Komeiji/i));
    expect(mockFn).toHaveBeenLastCalledWith('/koishi');

    fireEvent.click(await screen.findByText(/Hoshino Takanashi/i));
    expect(mockFn).toHaveBeenLastCalledWith('/hoshino');
  });
});
