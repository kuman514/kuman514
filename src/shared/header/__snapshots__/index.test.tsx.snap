// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Header should have appearances that matches snapshot 1`] = `
<header
  class="fixed z-header inset-x-0 top-0 w-full h-16 flex flex-row justify-between items-center bg-green-500 sm:px-6 md:px-8"
>
  <div
    class="h-full hidden md:flex md:flex-row md:justify-center md:items-center"
  >
    <a
      aria-label="kuman514-logo"
    >
      kuman514
    </a>
  </div>
  <nav
    class="w-full h-full md:w-fit"
  >
    <ul
      class="flex flex-row justify-between items-center w-full h-full md:gap-4"
    >
      <li
        id="yasuo"
      >
        <a>
          Yasuo
        </a>
      </li>
      <li
        id="koishi"
      >
        <a>
          Koishi Komeiji
        </a>
      </li>
      <li
        id="hoshino"
      >
        <a>
          Hoshino Takanashi
        </a>
      </li>
    </ul>
  </nav>
</header>
`;
